lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -244,6 +244,10 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
     {:noreply, finish_scan_binding(ref, state)}
   end
 
+  def handle_info({:orphan_log, output} = message, state) do
```

Comment:
Here, in `RuntimeServer`, add `:last_evaluator` to the state. It will be `nil` initially and then we set it when we evaluate code: https://github.com/livebook-dev/livebook/blob/8f4cafe264418c528b084563348b8f7ea74d5c9c/lib/livebook/runtime/erl_dist/runtime_server.ex#L334

This way we keep track of the process handling the most recent evaluation. Evaluator's group leader is our custom IOProxy process and it handles all the IO to get back to Livebook. So we can handle orphan log like this

```elixir
with %{} = evaluator <- state.last_evaluator,
     {:group_leader, io_proxy} <- Process.info(evaluator.pid, :group_leader) do
  # async_io to io_proxy
end
```

We could do `IO.puts(io_proxy, output)`, but we don't want to block (to receive IO response), so instead we can send the IO request manually and move on (the same way we do `async_io` in the custom logger backend).